### PR TITLE
Fixe #46

### PR DESCRIPTION
**Pull request by tluke900**
#46 

Added hardcoded loading of boneset names and drop down to select for them. Now only loads the selected boneset.

Drop down menu for bonesets and updated load button. 

Dynamic Loading work in progress, for future sets if added.

### CODE REVIEW

File: templates/boneset.html [hunk 1]
```diff
@@ -24,58 +24,103 @@
...
-                throw new Error('Network response was not ok');
-            }
             return response.json();
         }
 
+        async function populateDropdown() {
+            const data = await fetchJSON(BONESET_JSON_PATH);
+            const dropdown = document.getElementById('boneset-select');
+
+    // Clear existing options
```
Comment (Balijah) [inline templates/boneset.html:51]: Not sure what this is, is it necessary?

File: templates/boneset.html [hunk 2]
```diff
@@ -24,58 +24,103 @@
...
-                    boneSetList.appendChild(subList);
+        async function loadBoneSets() {
+        const dropdown = document.getElementById('boneset-select');
+        const selectedOption = dropdown.options[dropdown.selectedIndex];
+        const selectedBoneName = selectedOption.textContent;
+
+        try {
+            const bonesets = await fetchBonesetData();
+            console.log(bonesets);  // Check the fetched data
+
```
Comment (Balijah) [inline templates/boneset.html:94]: Good comments

**Comment by kate-holdener** [comment]
When I click "Load options" many times, I get a lot of duplicated information loaded.
Screenshot 2024-10-15 at 4 01 16 PM

**Review by jacksayshi — CHANGES_REQUESTED** [review]
Thanks for working on this.
The populateDropdown function that is adding options to the dropdown is getting called everytime load button is being clicked and this is addinf duplicate list of entries to the dropdown. 
We do not have to preserve the original buttons. Also we can try to fetch and populate dropdown while the page is loading (so that we dont call the function again). Here is what this issue solution could look like 
image

**Review by jacksayshi — APPROVED** [review]
We can merge these changes for now. We can make some more improvements like error handling and effective functions calls later.

**Pull request merged**